src/Servers/Kestrel/Core/src/Internal/Http3/Http3ControlStream.cs

```diff
@@ -318,23 +326,69 @@ private void ProcessSetting(long id, long value)
             }
         }
 
-        private ValueTask ProcessGoAwayFrameAsync()
+        private ValueTask ProcessGoAwayFrameAsync(ReadOnlySequence<byte> payload)
```

Comment:
`GOAWAY` is only sent from server.

> Clients do not need to send GOAWAY to initiate a graceful shutdown; they simply stop making new requests. A server MUST treat receipt of a GOAWAY frame on any stream as a connection error (Section 8) of type H3_FRAME_UNEXPECTED.

Reply:
(i see your code that is sending a H3_FRAME_UNEXPECTED to a bunch of streams; but it also looks like some additional processing is being done as if it were a valid frame. this should be a connection-level error and just kill everything)

Reply:
Yep, I misread the spec here. I thought that GOAWAY wasn't required but needed to be handled if sent. This makes it a lot easier on the Server side now.